### ISSUE TITLE
Refactor API handlers and vic-machine code to avoid duplication between CLI and API

### ISSUE THREAD

**Issue by zjs** [vmware/vic#6032]
Refactor vic-machine and API handlers code to avoid duplication between CLI and API

**Comment by zjs** [comment]
@hickeng has requested to be included on the design review for this work: https://github.com/vmware/vic/pull/7323#discussion_r168449634

**Comment by zjs** [comment]
- [ ] If this work includes making a manifest the primary means of defining a VCH configuration, consider https://github.com/vmware/vic/pull/7323#discussion_r168451137. (If not, ensure a separate issue is opened to track that eventual work, referencing this comment.)

**Comment by zjs** [comment]
Because this involves de-duplicating code, it should effectively increase test coverage; we'll be able to rely on _both_ API and CLI tests to verify the behavior of shared code. As such, it dovetails nicely with our Quality efforts.

**Comment by zjs** [comment]
Related: #7840

**Comment by renmaosheng** [comment]
moving to 'not ready' for further triage in VIC 1.5

**Comment by zjs** [comment]
> moving to 'not ready' for further triage in VIC 1.5
Is there specific information missing from this Epic which should be added before moving it to the backlog?

**Comment by renmaosheng** [comment]
> > moving to 'not ready' for further triage in VIC 1.5
> 
> Is there specific information missing from this Epic which should be added before moving it to the backlog?
No. As several Epics are not in the list from James, so I would like to have further triage for those Epics for VIC 1.5.
